app/process-batches/process-payment-file.js

```diff
@@ -0,0 +1,40 @@
+const blobStorage = require('../blob-storage')
+const batches = require('./batches')
+const reprocessIfNeeded = require('./reprocess-if-needed')
+const downloadAndParse = require('./download-and-parse')
+const { sendBatchErrorEvent } = require('../event')
+
+async function processPaymentFile (filename, schemeType) {
+  try {
+    console.log(`Processing ${filename}`)
+    const reprocessed = await reprocessIfNeeded(filename, schemeType)
+
+    if (!reprocessed) {
+      await checkSequenceAndPerfomAction(schemeType, filename)
```

Comment:
Still think the name can be approved.  `PerformAction` is a bit generic and doesn't really describe what is going to happen.  Also don't think `checkSequence` needs to be in the name as that's just a sub action of the function being called.  The calling function doesn't need to care about the details of that action.

The action is analysing the file and then either parsing it, ignoring it or quarantining it.  Not sure how best to describe that activity